server/database/models/CategoryRepository.js

```diff
@@ -0,0 +1,47 @@
+const AbstractRepository = require("./AbstractRepository"); 
+
+class CategoryRepository extends AbstractRepository{
+constructor() {
+    super({table : "category"});
+}
+
+async browse(){
+    const [result] = await this.database.query(`select name from ${this.table}`); 
+
+    return result; 
+}
+
+async read (id) {
+    const [rows] = await this.database.query(`select name from ${this.table} where id = ?`, [id]);
+
+    return rows[0]; 
+}
+
+async add (category) {
+    const [rows] = await this.database.query(
+        `insert into ${this.table} (name) value = ?`, 
+        [category.name]
```

Comment:
Where does `category.name` come from ?
You're passing `category` from `req.body`, there is no "name" inside (?)